src/baldr/graphtile.cc

```diff
@@ -80,44 +80,47 @@ GraphTile::GraphTile(const std::string& tile_dir, const GraphId& graphid) : head
       // read the compressed file into memory
       std::vector<char> compressed(std::istreambuf_iterator<char>(file),
                                    (std::istreambuf_iterator<char>()));
+      // try to decompress it
+      DecompressTile(graphid, compressed);
+    }
+  }
+}
 
-      // for setting where to read compressed data from
-      auto src_func = [&compressed](z_stream& s) -> void {
-        s.next_in = static_cast<Byte*>(static_cast<void*>(compressed.data()));
-        s.avail_in = static_cast<unsigned int>(compressed.size());
-      };
-
-      // for setting where to write the uncompressed data to
-      graphtile_.reset(new std::vector<char>(0, 0));
-      auto dst_func = [this, &compressed](z_stream& s) -> int {
-        // if the whole buffer wasn't used we are done
-        auto size = graphtile_->size();
-        if (s.total_out < size)
-          graphtile_->resize(s.total_out);
-        // we need more space
-        else {
-          // assume we need 3.5x the space
-          graphtile_->resize(size + (compressed.size() * COMPRESSION_HINT));
-          // set the pointer to the next spot
-          s.next_out = static_cast<Byte*>(static_cast<void*>(graphtile_->data() + size));
-          s.avail_out = compressed.size() * COMPRESSION_HINT;
-        }
-        return Z_NO_FLUSH;
-      };
-
-      // Decompress tile into memory
-      if (!baldr::inflate(src_func, dst_func)) {
-        LOG_WARN("Failed to gunzip: " + file_location + ".gz");
-        graphtile_.reset();
-        return;
-      }
+bool GraphTile::DecompressTile(const GraphId& graphid, std::vector<char>& compressed) {
```

Comment:
i have tests for this on the branch i just havent pushed t hem yet. need to implement `filesystem::remove_all` first